### ISSUE TITLE
Investigate JS bundle splitting

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons#11076]
There's some possible avenues to speed up the JS by splitting our code into smaller bundles. 

There's two distinct places to look at bundle splitting.

* Split out the vendor code into a separate bundle. We only get a gain from this if we don't change the vendor libs often - given the pace things move at we should consider if we would see much benefit from this. However it might make more sense in combination with the next item.
* Make multiple entry points in the app - this would mean that a specific page would load only the relevant JS for itself rather than loading the JS for the whole site. 

A new issue should be filed for each of these so they can be looked at independently.

This area could have some trade-offs so we should make sure that if we try them, we do each one in a way that allows us to measure the effects of implementing them and we should investigate what others have found in this area before we start.

_Update: Additional considerations_

Another consideration here is to think about the different types of users. On a desktop with a fast connection the big bundle up front might be better, but for a mobile client it could be a completely different story.

We should investigate traffic and usage patterns to see if this is something that we'd be likely to benefit from.

---

**Update for QA:** the about and review guides pages are loaded differently now:

- https://addons-dev.allizom.org/en-US/about
- https://addons-dev.allizom.org/en-US/review_guide

* There should be no visual regression and no issues when navigating to/from these pages.
* There should be NO page reload when loading one of these pages and then clicking another (internal) link in the menu (or the FF logo).
* You _might_ notice a short delay on FF when clicking the "about" link from the homepage for example, but that should not impact the user experience and there should be absolutely no delay in Chrome. I'll file a separate issue for that.
